sonar-helm-for-iac/converters/tree_converter.go

```diff
@@ -79,208 +103,234 @@ func convert(node parse.Node) proto.Message {
 			Pos:      int64(node.Position()),
 		}
 	}
+
 	return nodeAsMessage
 }
 
-func convertActionNode(node parse.ActionNode) proto.Message {
+func (c ConversionContext) convertActionNode(node parse.ActionNode) proto.Message {
 	return &pbstructs.ActionNode{
 		NodeType: pbstructs.NodeType_NodeAction,
-		Pos:      int64(node.Pos),
-		Pipe:     convertPipeNode(*node.Pipe),
+		Pos:      int64(node.Position()),
+		Pipe:     c.convertPipeNode(*node.Pipe),
+		Length:   nodeSourceLength(&node),
 	}
 }
 
-func convertBoolNode(node parse.BoolNode) proto.Message {
+func (c ConversionContext) convertBoolNode(node parse.BoolNode) proto.Message {
+	var length int
+	if node.True {
+		length = len(KwTrue)
+	} else {
+		length = len(KwFalse)
+	}
 	return &pbstructs.BoolNode{
 		NodeType: pbstructs.NodeType_NodeBool,
-		Pos:      int64(node.Pos),
+		Pos:      int64(node.Position()),
 		True:     node.True,
+		Length:   int64(length),
 	}
 }
 
-func convertBranchNode(node parse.BranchNode) proto.Message {
+func (c ConversionContext) convertBranchNode(node parse.BranchNode) proto.Message {
 	var elseList *pbstructs.ListNode
 	if node.ElseList == nil {
 		elseList = nil
 	} else {
-		elseList = convertListNode(*node.ElseList)
+		elseList = c.convertListNode(*node.ElseList)
 	}
 	return &pbstructs.BranchNode{
-		Pos:      int64(node.Pos),
-		Pipe:     convertPipeNode(*node.Pipe),
-		List:     convertListNode(*node.List),
+		Pos:      int64(node.Position()),
+		Pipe:     c.convertPipeNode(*node.Pipe),
+		List:     c.convertListNode(*node.List),
 		ElseList: elseList,
+		Length:   nodeSourceLength(&node),
 	}
 }
 
-func convertBreakNode(node parse.BreakNode) proto.Message {
+func (c ConversionContext) convertBreakNode(node parse.BreakNode) proto.Message {
 	return &pbstructs.BreakNode{
 		NodeType: pbstructs.NodeType_NodeBreak,
-		Pos:      int64(node.Pos),
+		Pos:      int64(node.Position()),
+		Length:   int64(len(KwBreak)),
 	}
 }
 
-func convertChainNode(node parse.ChainNode) proto.Message {
-	return &pbstructs.ChainNode{
```

Comment:
I still can't understand if these nodes can actually occur or are just legacy. Because of that, I can't properly test it, so I decided to remove it.

Reply:
Nope, it appears in things like `(.Values).foo`. Added a test case and restored code